### PR TITLE
[coqdoc] Fix #11353: coqdoc -g omits all sentences with decorations

### LINKED ISSUE #11353: coqdoc with -g option omits all sentences with decorations
#### Description of the problem

Consider the following example file, `mysum.v`, adapted from `Datatypes.v` in the stdlib:
```coq
#[universes(template)]
Inductive mysum (A B:Type) : Type :=
  | myinl : A -> mysum A B
  | myinr : B -> mysum A B.
```
Now, assume we run `coqdoc` with the `-g` option (used when generating the official HTML documentation):
```bash
coqdoc --html -g mysum.v
```
This results in a HTML file where the `mysum` definition is missing:
```html
<div id="main">

<h1 class="libtitle">Library mysum</h1>

<div class="code">
</div>
</div>
```
This is not restricted to HTML, e.g., the following gives a PDF without the `mysum` definition:
```bash
coqdoc --pdf -g mysum.v -o mysum.pdf
```
Note that this is restricted to inductives with `#[universes(template)]` annotations, and dropping the `-g` option makes everything work as expected.

The absence of these annotated inductives makes it difficult to read the stdlib documentation for Coq 8.10, and invalidates many links in generated HTML documentation.

For example, here is the URL failing to pinpoint the definition of `sum`: https://coq.inria.fr/distrib/V8.10.2/stdlib/Coq.Init.Datatypes.html#sum
This URL was actually fully functional in the 8.9 documentation: https://coq.inria.fr/distrib/V8.9.0/stdlib/Coq.Init.Datatypes.html#sum

#### Coq Version

This affects `coqdoc` in all released 8.10 versions, 8.11beta1 and master.

### PR DESCRIPTION
**Pull request by palmskog**
**Kind:** bug fix

Fixes #11353

- [x] Added / updated test-suite

- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

**Comment by palmskog** [comment]
@ppedrot I'm going through the test suite for coqdoc, and it looks like the `-g` flag isn't used there when calling any of the test cases, despite it being used to produce stdlib documentation. Can you advise whether I should add the flag in some way? Otherwise, the bug will not be triggered for sure...

**Comment by ppedrot** [comment]
Yes, please add the `-g` flag to the test. Reading `test-suite/Makefile`, it seems it is enough to add the invocation
```
(* -*- coq-prog-args: ("-g") -*- *)
```
at the beginning of the file.

**Review by herbelin — APPROVED** [review]
It looks to be the right fix to me. `coqdoc -g` is looking for sentences starting with a keyword and everything not starting with a known keyword is supposed to be a tactic and is skipped up to the next period (`.`).

Note that this affects several new commands not yet registered in coqdoc, like e.g. `Arguments`.

So, this PR fixes the case of `#[ ... ]` but a couple of possible start of sentences is still missing in `coqdoc`.

In passing, `Ltac2` is luckily recognized because `Ltac` is recognized but the `2` is not hightlighted. Other commands skipped include `Grab ...`, `Derive ...`.

Also, virtually all commands defined in plugins and not starting with an existing keyword are skipped too by `coqdoc -g`.

File: doc/changelog/08-tools/11394-fix-coqdoc-annotations.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- **Fixed:**
+  ``coqdoc`` with option ``-g`` (Gallina only) omits all sentences with decorations
```
Comment (Zimmi48) [inline doc/changelog/08-tools/11394-fix-coqdoc-annotations.rst]: The use of present tense is ambiguous: it looks like what you describe is the new (correct) behavior. What about:

>coqdoc now supports commands with attributes

**Comment by Zimmi48** [comment]
@herbelin Do you self-assign? I think this should be merged promptly because it is one of the last remaining blockers for 8.11.0.

**Comment by palmskog** [comment]
I think all reviewing concerns are addressed now (e.g., changelog should not be ambiguous anymore), and all CI failures appear spurious.

**Comment by herbelin** [comment]
> Do you self-assign?

OK, I assign. Going to merge soon.

**Pull request merged**